src/Chapter3.hs

```diff
@@ -480,6 +520,50 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
 
+data Castle = Castle 
+  { castleName :: String
+  , hasWalls :: Bool
+  } deriving (Show)
+
+makeCastle :: String -> Bool -> Castle
+makeCastle n withWalls = Castle { castleName = n, hasWalls = True }
+
+data PersonCount = One | Two | Three | Four deriving (Show)
+data House = House PersonCount deriving (Show)
+
+houseCount :: House -> Int
+houseCount (House One) = 1
+houseCount (House Two) = 2
+houseCount (House Three) = 3
+houseCount (House Four) = 4
```

Comment:
To avoid some code duplication, you can use the `case-of` expression in such cases:

```haskell
houseCount (House h) = case h of
    One -> ...
    ... -> ...
```

Reply:
Got it. Implemented your suggestion. Thanks 😊